tests/py/test_billing_payday.py

```diff
@@ -568,6 +492,138 @@ def test_payin_dumps_transfers_for_debugging(self, cch, fch):
+        self.start_payday()
+        self.make_member('bruiser', 250)
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('  0.00')
+        assert P('picard').balance  == D('  0.00')
+
+    def test_pt_ignores_takes_that_have_already_been_processed(self):
+        self.make_member('crusher', 150)
+        self.start_payday()
+        self.make_member('bruiser', 250)
+        self.run_through_takes()
+        self.run_through_takes()
+        self.run_through_takes()
+        self.run_through_takes()
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('  0.00')
+        assert P('picard').balance  == D('  0.00')
+
+    def test_pt_clips_to_available(self):
+        self.make_member('alice', 350)
+        self.make_member('bruiser', 250)
+        self.make_member('crusher', 150)
+        self.make_member('zorro', 450)
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('250.00')
+        assert P('alice').balance   == D('100.00')
+        assert P('zorro').balance   == D('  0.00')
+        assert P('picard').balance  == D('  0.00')
+
+    def test_pt_clips_to_balance_when_less_than_available(self):
+        self.set_available(1000)
+        self.make_member('alice', 350)
+        self.make_member('bruiser', 250)
+        self.make_member('crusher', 150)
+        self.make_member('zorro', 450)
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('250.00')
+        assert P('alice').balance   == D('137.00')
+        assert P('zorro').balance   == D('  0.00')
+        assert P('picard').balance  == D('  0.00')
+
+    def test_pt_is_NOT_happy_to_deal_the_owner_in(self):
+        self.make_member('crusher', 150)
+        self.make_member('bruiser', 250)
+        self.enterprise.set_take_for(self.picard, D('0.01'), self.picard)
+        with pytest.raises(NotAllowed):
+            self.enterprise.set_take_for(self.picard, 50, self.picard)
+        return  # XXX allow owners to join teams!
+        self.run_through_takes()
+        assert P('crusher').balance == D('150.00')
+        assert P('bruiser').balance == D('250.00')
+        assert P('picard').balance  == D(' 50.00')
+
+    def test_process_remainder_only_processes_pt_leftovers(self):
+        self.make_member('alice', 100)
+        self.start_payday()
+        with self.db.get_cursor() as cursor:
+            self.payday.prepare(cursor)
+            cursor.run("UPDATE payday_teams SET balance=500")
+            self.payday.process_takes(cursor, self.payday.ts_start)
+            self.payday.process_remainder(cursor)
+            self.payday.update_balances(cursor)
+
+        assert P('alice').balance == D('100')
+        assert P('picard').balance == D('400') # Owner receives whatever is left
```

Comment:
Is this test expected to fail without d971d40d4f29ac4fb8d4e955928b1194d7e642a3? That was my expectation, but when I manually revert that commit locally this test still passes.

Also: Let's find the right place for this in the test suite. Where it is now it should have a name like `test_pt_` for both conceptual grouping and easy filtering with `py.test -k test_pt_`. Maybe it would work better up in the `TestPayday` class?